File: client-app/create_assets.py
import os
import base64

assets_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'assets')
os.makedirs(assets_dir, exist_ok=True)

# A small 1x1 solid dark-blue PNG file to prevent Expo asset loading crashes
base64_png = (
    'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII='
)
png_data = base64.b64decode(base64_png)

filenames = ['icon.png', 'splash.png', 'adaptive-icon.png', 'favicon.png']
for name in filenames:
    filepath = os.path.join(assets_dir, name)
    with open(filepath, 'wb') as f:
        f.write(png_data)
    print(f"Created asset: {filepath}")

File: client-app/create_assets.js
const fs = require('fs');
const path = require('path');

const assetsDir = path.join(__dirname, 'assets');
if (!fs.existsSync(assetsDir)){
    fs.mkdirSync(assetsDir, { recursive: true });
}

const base64Png = 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=';
const pngData = Buffer.from(base64Png, 'base64');

const filenames = ['icon.png', 'splash.png', 'adaptive-icon.png', 'favicon.png'];
filenames.forEach(name => {
    const filepath = path.join(assetsDir, name);
    fs.writeFileSync(filepath, pngData);
    console.log(`Created asset: ${filepath}`);
});
